Add "garden" - a hacker-friendly multi-repo Git power tool

davvid opened the pull request:
Garden is a uniquely powerful Git-centric tool that makes it easy to perform dynamic operations over arbitrary collections of configuration-defined Git trees.

https://gitlab.com/garden-rs/garden

https://garden-rs.gitlab.io/

https://crates.io/crates/garden-tools

davvid committed:
Add "garden" - a hacker-friendly multi-repo Git power tool

Garden is a uniquely powerful Git-centric tool that makes it easy to
perform dynamic operations over arbitrary collections of
configuration-defined Git trees.

https://gitlab.com/garden-rs/garden

https://garden-rs.github.io/

https://crates.io/crates/garden-tools

dictcp commented:
I am not sure what `garden` is. maybe it is more like a workflow / CICD tools? 🤔

davvid commented:
Hi @dictcp ~ here's a few use cases in which garden is useful. It's very much a git-centric helper.

Do you ever find yourself with random collections of Git repos that you'd like to index and catalog in a shareable config file?

Do you ever find yourself wanting to recreate these collections of Git repositories from scratch?

Do you ever find yourself collecting random collections of shell scripts and commands snippets alongside these catalogs of Git repositories?


These are the basic use cases that garden solves. I was able to replace lots of random shell scripts and build instructions with a single garden file where I can keep notes and custom command snippets.

I hope that helps explain some of the motivation garden use cases.

Thanks for responding to this PR!

koppor commented:
https://garden-rs.github.io/ is 404 @davvid

davvid commented:
Sorry 'bout that, we moved to gitlab. https://garden-rs.gitlab.io/

I'll try to update this PR tonight/tomorrow.

Update: nevermind, there's nothing to update in the branch as it's already linked to the gitlab repo. We should be all good here. I updated the link in the description above. Sorry 'bout that.
